### DISCUSSION TITLE
Failed to retrieve security status update for '4.8.3'

### DISCUSSION THREAD

**Discussion by ballou-git** [PowerDNS/pdns#13466] [Q&A]
Hi, I hope I post in the right place.
Since we have updated all our PowerDNS setups and made them work perfect we still find this error.
`Failed to retrieve security status update for '4.8.3' on 'auth-4.8.3.security-status.secpoll.powerdns.com.': RCODE was Server Failure`
Can someone please tell me where the error is?
I'm told that it is on you part, can this be true?

Best regards
Fredrik

**Comment by Habbie** [comment]
can you try `dig txt auth-4.8.3.security-status.secpoll.powerdns.com.` on a machine that has this problem, and show us the full output?

**Comment by ballou-git** [comment]
Thank you, this is the output and I've removed the IP of our resolver at the end
```
# dig txt auth-4.8.3.security-status.secpoll.powerdns.com

; <<>> DiG 9.16.23-RH <<>> txt auth-4.8.3.security-status.secpoll.powerdns.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 63379
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;auth-4.8.3.security-status.secpoll.powerdns.com. IN TXT

;; ANSWER SECTION:
auth-4.8.3.security-status.secpoll.powerdns.com. 60 IN TXT "1 OK"

;; Query time: 80 msec
;; SERVER: XXX.XXX.XXX.XXX#53(XXX.XXX.XXX.XXX)
;; WHEN: Tue Nov 14 08:16:18 CET 2023
;; MSG SIZE  rcvd: 93
```

**Comment by ballou-git** [comment]
No help here? I would really like to get rid of this error.

**Reply by rgacogne** [reply]
The output doesn't show any problem so it's likely that your resolver is failing to resolve from time to time, but not always, which is something you will have to investigate on your resolver's side.

**Comment by ballou-git** [comment]
Thank you.
There is no A-record we get no IP.
You can't just use a TXT record without an A-record as what I know.

**Reply by rgacogne** [reply]
There is no need for a A record, having only type for a given name is perfectly fine. This is not the issue. The `Failed to retrieve security status update for '4.8.3' on 'auth-4.8.3.security-status.secpoll.powerdns.com.': RCODE was Server Failure` message is stating that the Authoritative server got a `Server Failure` response from your resolver when asking for the `TXT` record at `auth-4.8.3.security-status.secpoll.powerdns.com.`, which is not a permanent failure since sending the same query with `dig` succeeded. Finding out why your resolver sent this `Server Failure` response will have to be investigated on the resolver, as there are a lot of potential explanations (overloaded server, network connectivity issue, transient resolution failure from the servers authoritative for `secpoll.powerdns.com.'`, ...) and only the resolver knows why it sent a `Server Failure`.

**Reply by ballou-git** [reply]
It must send an A record. We have no problems with overload or connection.
The problem is that our server does not know where to route because it sends a TXT record.
Then it is impossible for our servers to resolve an IP address.
The TXT record does not contain an IP address.
No IP address = no routing.

**Reply by rgacogne** [reply]
Sorry, what you are saying does not make any sense to me. Our security polling works by requesting a TXT record via a standard DNS query, there is no IP address involved.

**Reply by ballou-git** [reply]
I really don't see how the resolver can work without an IP?
Can you please guide me?

**Reply by rgacogne** [reply]
I would suggest reading how DNS resolution works; a good start might be https://en.wikipedia.org/wiki/Domain_Name_System#Address_resolution_mechanism

**Reply by ballou-git** [reply]
I'm sorry but I really don't think you are serious.
The last post proves it.

**Reply by Habbie** [reply]
hello @ballou-git, this is not appropriate and not acceptable. We are serious, we are trying to help you, and we are putting in effort. Please adjust your behaviour, or leave.

**Comment by omoerbeek** [comment]
For people reading this question some information on how the authoritative servers retrieves the secpoll status:

The autoritatieve server uses the value of the `resolver` setting to send queries to if it needs to resolve names (outside the zones it's authoritative for). If `resolver` is not set, the system's `/etc/resolv.conf` is parsed and the nameservers mentioned in it are used. The addresses found are used tho query a resolver to resolve the secpoll name to a TXT record that indicates the status. See also https://docs.powerdns.com/authoritative/security.html#security-polling. If you log at `debug` level, you can follow the steps.

Additionally: `dnsdist` only uses the addresses in `/etc/resolv.conf` and  the recursor resolves the name itself.
